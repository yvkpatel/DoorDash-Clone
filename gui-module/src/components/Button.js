const Button = ({text, color, onClick}) => {
    return(
        <button className='button'
        style={{background:color}}
        data-testid="button"
        onClick={onClick}>
            {text}
        </button>
    )
}

export default Button