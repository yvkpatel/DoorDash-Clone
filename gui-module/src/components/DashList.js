/*  DashList.js
    Component that acts as a container for cards displayed in a veritcal list.

    === Props ===
    itemList = list of the items to be displayed in each of the cards.
        Example: const [listVar] = useState([{ id: 1, title: 'Card 1'}, { id: 2, title: 'Card 2'}]);

    component = the component that will be used for each of the cards.

    manager = class that will provide methods that may be called from the card.
*/

const List = ({ itemList, component, manager }) => {
  const Component = component;

  return (
    <div className="list" data-testid="dash-list">
      {itemList
        ? itemList.map((item) => (
            <Component key={item.id} item={item} manager={manager} />
          ))
        : "Loading:..."}
    </div>
  );
};

export default List;
