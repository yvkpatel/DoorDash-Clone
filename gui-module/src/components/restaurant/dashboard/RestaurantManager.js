/*  RestaurantManager.js
    Provides methods for the restaurant dashboard to interact with databases and manage the restaurant's local information.
*/
class RestaurantManager {
  constructor(id, income, orders) {
    this.id = id;
    this.income = income;
    this.orders = orders;
  }

  // =============== LOCAL MANAGEMENT METHODS ===============

  // Converts the restaurant's income from a float to a currency formatted string.
  getIncome() {
    return "$" + this.income;
  }

  updateIncome(x) {
    this.income += x;
  }

  // =============== METHODS THAT INTERACT WITH DATABASES ===============

  // PLACEHOLDER: will fetch restaurant's orders from database.
  // Current implementation simply returns a hardcoded list of imaginary orders.
  getRestaurantOrders() {
    return [
      {
        id: 1,
        title: "Presidential Pizza",
        restaurant: "Boston Pizza",
        destination: "1600 Pennsylvania Avenue",
      },
      {
        id: 2,
        title: "Famous Pizza, I guess.",
        restaurant: "Dominos",
        destination: "4059 Mt Lee Dr. Hollywood",
      },
      {
        id: 3,
        title: "Pentagram Pizza",
        restaurant: "Pizza Hut",
        destination: "6114 California Street",
      },
      {
        id: 4,
        title: "Curious Pizza",
        restaurant: "???",
        destination: "221B Baker Street",
      },
      {
        id: 5,
        title: "Pizza, dude.",
        restaurant: "Literally Anywhere",
        destination: "122 + 1/8th Bleaker Street",
      },
    ];
  }

  // PLACEHOLDER: When called it should inform database that the order status has changed.
  updateOrder(orderID) {
    console.log("Order ID: " + orderID);
  }
}

export default RestaurantManager;
