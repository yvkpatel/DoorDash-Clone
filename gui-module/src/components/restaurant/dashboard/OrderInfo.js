// Displays customer's order info to restauraants
import React from "react";
import Button from "../../Button";

// TEMPORARILY USES THE SAME STYLING AS DriverOrder.js
// TODO: Renaming CSS classes to be more generic.

const OrderInfo = (props) => {
  return (
    <div className="driverTask">
      <div className="driverTaskHeader">
        <h2>{props.item.title}</h2>
        <Button
          text="Complete"
          color="#2DB6F0"
          onClick={() => props.manager.updateOrder(props.item.id)}
        />
      </div>
      <div className="driverTaskDetails">
        <p>Order Comments: {props.item.comments}</p>
      </div>
    </div>
  );
};

export default OrderInfo;
