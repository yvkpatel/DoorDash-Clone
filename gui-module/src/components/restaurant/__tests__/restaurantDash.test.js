import { render, screen, cleanup, fireEvent } from "@testing-library/react";
import renderer from "react-test-renderer";
import RestaurantDash from "../dashboard/RestaurantDashPage";
import userEvent from "@testing-library/user-event";
import { BrowserRouter } from "react-router-dom";
import ReactDOM from "react-dom";
import Button from "../../Button";
import DashList from "../../DashList";
import OrderInfo from "../dashboard/OrderInfo";
import RestaurantManager from "../dashboard/RestaurantManager";
import React, { useEffect } from "react";
import "@testing-library/jest-dom/extend-expect";

afterEach(() => {
  cleanup();
});

test("should render restaurant dashboard component", () => {
  render(
    <BrowserRouter>
      <RestaurantDash />
    </BrowserRouter>
  );
  const RegisterResElement = screen.queryByTestId("register-res");
  expect(RegisterResElement).toBeDefined();
});

test("Complete button renders", () => {
  render(
    <BrowserRouter>
      <RestaurantDash />
    </BrowserRouter>
  );
  const DriverDashButtons = screen.getAllByRole("button", { name: "Complete" });
  DriverDashButtons.forEach((element) => {
    expect(element).toBeEnabled();
  });
});

const MockList = [
  {
    id: 4,
    title: "Test Pizza",
    restaurant: "???",
    destination: "221B Baker Street",
  },
];

test("List of orders renders", () => {
  let manager = new RestaurantManager(27, 9689.25, 212);
  const { getByText } = render(
    <DashList itemList={MockList} component={OrderInfo} manager={manager} />
  );

  expect(getByText("Test Pizza")).toBeTruthy();
});

test("matches snapshot", () => {
  const tree = renderer.create(
    <BrowserRouter>
      <RestaurantDash />
    </BrowserRouter>
  );
  expect(tree).toMatchSnapshot();
});
