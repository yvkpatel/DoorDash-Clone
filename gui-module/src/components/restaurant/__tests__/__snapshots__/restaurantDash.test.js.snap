// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="App"
  datatest-id="register-res"
>
  <header
    className="header"
  >
    <img
      alt="YUMI Logo"
      className="logo"
      src="logo-header.png"
    />
    <a
      className="accSettings"
      href="https://developer.mozilla.org/en-US/docs/Web/CSS"
    >
      Account Settings
    </a>
    <a
      href="/"
      onClick={[Function]}
    >
      <button
        className="button"
        data-testid="button"
        onClick={[Function]}
        style={
          Object {
            "background": "#2DB6F0",
          }
        }
      >
        Logout
      </button>
    </a>
  </header>
  <div
    className="driverDashMain"
  >
    <div
      className="dashInfoBox"
    >
      <h1>
        Account Summary
      </h1>
      <p>
        Income: 
        $9689.25
      </p>
      <p>
        Orders: 
        212
      </p>
    </div>
    <div
      className="list"
      data-testid="dash-list"
    >
      <div
        className="driverTask"
      >
        <div
          className="driverTaskHeader"
        >
          <h2>
            Presidential Pizza
          </h2>
          <button
            className="button"
            data-testid="button"
            onClick={[Function]}
            style={
              Object {
                "background": "#2DB6F0",
              }
            }
          >
            Complete
          </button>
        </div>
        <div
          className="driverTaskDetails"
        >
          <p>
            Order Comments: 
          </p>
        </div>
      </div>
      <div
        className="driverTask"
      >
        <div
          className="driverTaskHeader"
        >
          <h2>
            Famous Pizza, I guess.
          </h2>
          <button
            className="button"
            data-testid="button"
            onClick={[Function]}
            style={
              Object {
                "background": "#2DB6F0",
              }
            }
          >
            Complete
          </button>
        </div>
        <div
          className="driverTaskDetails"
        >
          <p>
            Order Comments: 
          </p>
        </div>
      </div>
      <div
        className="driverTask"
      >
        <div
          className="driverTaskHeader"
        >
          <h2>
            Pentagram Pizza
          </h2>
          <button
            className="button"
            data-testid="button"
            onClick={[Function]}
            style={
              Object {
                "background": "#2DB6F0",
              }
            }
          >
            Complete
          </button>
        </div>
        <div
          className="driverTaskDetails"
        >
          <p>
            Order Comments: 
          </p>
        </div>
      </div>
      <div
        className="driverTask"
      >
        <div
          className="driverTaskHeader"
        >
          <h2>
            Curious Pizza
          </h2>
          <button
            className="button"
            data-testid="button"
            onClick={[Function]}
            style={
              Object {
                "background": "#2DB6F0",
              }
            }
          >
            Complete
          </button>
        </div>
        <div
          className="driverTaskDetails"
        >
          <p>
            Order Comments: 
          </p>
        </div>
      </div>
      <div
        className="driverTask"
      >
        <div
          className="driverTaskHeader"
        >
          <h2>
            Pizza, dude.
          </h2>
          <button
            className="button"
            data-testid="button"
            onClick={[Function]}
            style={
              Object {
                "background": "#2DB6F0",
              }
            }
          >
            Complete
          </button>
        </div>
        <div
          className="driverTaskDetails"
        >
          <p>
            Order Comments: 
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
